File: DataTransferObjects/ProductDesignDTOs/UpdateProductDesignDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataTransferObjects.ProductDesignDTOs
{
    public class UpdateProductDesignDTO
    {
        public Guid UserId { get; set; }
        public Guid BlankVarianceId { get; set; }
        public string? Saved3DPreviewUrl { get; set; }
        public bool? IsFinalized { get; set; }   // trong mô tả bạn ghi "Không có", nhưng nếu bạn cần update thì thêm
        public bool? IsPublic { get; set; }
        public bool? IsTemplate { get; set; }
    }
}

File: src/product-design/dto/update-product-design.dto.ts
import { Field, ID, InputType } from '@nestjs/graphql';

@InputType()
export class UpdateProductDesignDto {
  @Field(() => ID)
  id: string;

  @Field({ nullable: true })
  isFinalized?: boolean;

  @Field({ nullable: true })
  isPublic?: boolean;

  @Field({ nullable: true })
  isTemplate?: boolean;
} 